examples/pytorch/grand/main.py

```diff
@@ -0,0 +1,198 @@
+import dgl.function as fn
+def GRANDConv(graph, feats, order):
+    
+    with graph.local_scope():
+        
+        degs = graph.in_degrees().float().clamp(min=1)
+        norm = th.pow(degs, -0.5).to(feats.device).unsqueeze(1)
+
+        graph.ndata['norm'] = norm
+        graph.apply_edges(fn.u_mul_v('norm', 'norm', 'weight'))
+
+        x = feats
+        y = 0+feats
+
+        for i in range(order):
+            graph.ndata['h'] = x
+            graph.update_all(fn.u_mul_e('h', 'weight', 'm'), fn.sum('m', 'h'))
+            x = graph.ndata.pop('h')
+            y.add_(x)
+
+    return x
+
+
+import argparse
+import numpy as np
+import torch as th
+import torch.optim as optim
+import torch.nn as nn
+import torch.nn.functional as F
+
+import dgl
+from dgl.data import CoraGraphDataset, CiteseerGraphDataset, PubmedGraphDataset
+
+from model import GRAND
+from utils import consis_loss
+
+import warnings
+warnings.filterwarnings('ignore')
+
+def argument():
+
+    parser = argparse.ArgumentParser(description='GRAND')
+
+    # data source params
+    parser.add_argument('--dataname', type=str, default='cora', help='Name of dataset.')
+    # cuda params
+    parser.add_argument('--gpu', type=int, default=-1, help='GPU index. Default: -1, using CPU.')
+    # training params
+    parser.add_argument('--epochs', type=int, default=200, help='Training epochs.')
+    parser.add_argument('--early_stopping', type=int, default=200, help='Patient epochs to wait before early stopping.')
+    parser.add_argument('--lr', type=float, default=0.01, help='Learning rate.')
+    parser.add_argument('--weight_decay', type=float, default=5e-4, help='L2 reg.')
+    # model params
+    parser.add_argument("--hid_dim", type=int, default=32, help='Hidden layer dimensionalities.')
+    parser.add_argument('--dropnode_rate', type=float, default=0.5,
+                        help='Dropnode rate (1 - keep probability).')
+    parser.add_argument('--input_droprate', type=float, default=0.0,
+                    help='dropout rate of input layer')
+    parser.add_argument('--hidden_droprate', type=float, default=0.0,
+                    help='dropout rate of hidden layer')
+    parser.add_argument('--order', type=int, default=8, help='Propagation step')
+    parser.add_argument('--sample', type=int, default=4, help='Sampling times of dropnode')
+    parser.add_argument('--tem', type=float, default=0.5, help='Sharpening temperature')
+    parser.add_argument('--lam', type=float, default=1., help='Lamda')
+    parser.add_argument('--use_bn', action='store_true', default=False, help='Using Batch Normalization')
+
+    args = parser.parse_args()
+    if args.gpu != -1 and th.cuda.is_available():
+        args.device = 'cuda:%s'%args.gpu
+    else:
+        args.device = 'cpu'
+
+    return args
+
+if __name__ == '__main__':
+
```

Comment:
Please refer to the MVPExampleCode's main.py to add some comments of your main function.